examples/tokens/ERC4626.vy

```diff
@@ -0,0 +1,257 @@
+def transferFrom(sender: address, receiver: address, amount: uint256) -> bool:
+    self.allowance[sender][msg.sender] -= amount
+    self.balanceOf[sender] -= amount
+    self.balanceOf[receiver] += amount
+    log Transfer(sender, receiver, amount)
+    return True
+
+
+@view
+@external
+def totalAssets() -> uint256:
+    return self.asset.balanceOf(self)
+
+
+@view
+@internal
+def _convertToAssets(shareAmount: uint256) -> uint256:
+    totalSupply: uint256 = self.totalSupply
+    if totalSupply == 0:
+        return 0
+
+    # NOTE: `shareAmount = 0` is extremely rare case, not optimizing for it
+    # NOTE: `totalAssets = 0` is extremely rare case, not optimizing for it
+    return shareAmount * self.asset.balanceOf(self) / totalSupply
+
+
+@view
+@external
+def convertToAssets(shareAmount: uint256) -> uint256:
+    return self._convertToAssets(shareAmount)
+
+
+@view
+@internal
+def _convertToShares(assetAmount: uint256) -> uint256:
+    totalSupply: uint256 = self.totalSupply
+    totalAssets: uint256 = self.asset.balanceOf(self)
+    if totalAssets == 0 or totalSupply == 0:
+        return assetAmount  # 1:1 price
+
+    # NOTE: `assetAmount = 0` is extremely rare case, not optimizing for it
+    return assetAmount * totalSupply / totalAssets
+
+
+@view
+@external
+def convertToShares(assetAmount: uint256) -> uint256:
+    return self._convertToShares(assetAmount)
+
+
+@view
+@external
+def maxDeposit(owner: address) -> uint256:
+    return MAX_UINT256
+
+
+@view
+@external
+def previewDeposit(assets: uint256) -> uint256:
+    return self._convertToShares(assets)
+
+
+@external
+def deposit(assets: uint256, receiver: address=msg.sender) -> uint256:
+    shares: uint256 = self._convertToShares(assets)
+    self.asset.transferFrom(msg.sender, self, assets)
+
+    self.totalSupply += shares
+    self.balanceOf[receiver] += shares
+    log Deposit(msg.sender, receiver, assets, shares)
+    return shares
+
+
+@view
+@external
+def maxMint(owner: address) -> uint256:
+    return MAX_UINT256
+
+
+@view
+@external
+def previewMint(shares: uint256) -> uint256:
+    assets: uint256 = self._convertToAssets(shares)
+
+    # NOTE: Vyper does lazy eval on if, so this avoids SLOADs most of the time
+    if assets == 0 and self.asset.balanceOf(self) == 0:
+        return shares  # NOTE: Assume 1:1 price if nothing deposited yet
+
+    return assets
+
+
+@external
+def mint(shares: uint256, receiver: address=msg.sender) -> uint256:
+    assets: uint256 = self._convertToAssets(shares)
+
+    if assets == 0 and self.asset.balanceOf(self) == 0:
+        assets = shares  # NOTE: Assume 1:1 price if nothing deposited yet
+
+    self.asset.transferFrom(msg.sender, self, assets)
+
+    self.totalSupply += shares
+    self.balanceOf[receiver] += shares
+    log Deposit(msg.sender, receiver, assets, shares)
+    return assets
+
+
+@view
+@external
+def maxWithdraw(owner: address) -> uint256:
+    return MAX_UINT256  # real max is `self.asset.balanceOf(self)`
```

Comment:
does that mean this is a stub?

Reply:
Yes, but one we expect per the standard. This is useful in certain more complex scenarios like withdrawal limits